app/controllers/WhatIsThePropertyOrLandAddressController.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import controllers.filters.IndexActionFilterProvider
+import forms.UKAddressFormProvider
+import javax.inject.Inject
+import models.Mode
+import navigation.Navigator
+import pages.WhatIsThePropertyOrLandAddressPage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import uk.gov.hmrc.play.bootstrap.controller.FrontendBaseController
+import views.html.WhatIsThePropertyOrLandAddressView
+import mapping.reads.Assets
+import scala.concurrent.{ExecutionContext, Future}
+
+class WhatIsThePropertyOrLandAddressController @Inject()(
+                                        override val messagesApi: MessagesApi,
+                                        sessionRepository: SessionRepository,
+                                        navigator: Navigator,
+                                        identify: IdentifierAction,
+                                        getData: DraftIdRetrievalActionProvider,
+                                        requireData: DataRequiredAction,
+                                        validateIndex: IndexActionFilterProvider,
+                                        formProvider: UKAddressFormProvider,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        view: WhatIsThePropertyOrLandAddressView
+                                    )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  val form = formProvider()
+
+  private def actions(index : Int, draftId: String) =
+    identify andThen getData(draftId) andThen
+      requireData andThen
+      validateIndex(index, mapping.reads.Assets)
```

Comment:
This should be sections.Assets not readss?

Reply:
Done